src/cargo/ops/cargo_new.rs

```diff
@@ -17,11 +17,13 @@ use toml;
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum VersionControl { Git, Hg, NoVcs }
 
+#[derive(Clone, Debug, PartialEq)]
 pub struct NewOptions<'a> {
     pub version_control: Option<VersionControl>,
     pub bin: bool,
     pub path: &'a str,
     pub name: Option<&'a str>,
+    pub sourcefile_relative_path : Option<String>,
```

Comment:
Yeah that seems plausible, it could also just be an extra parameter to `mk` or could be attached to a `MkContext` which has a method `doit` or something like that.